### ISSUE TITLE
baremetal: Migrate Ironic services to upstream images 

### ISSUE THREAD

**Issue by hardys** [openshift/installer#2090]
The integration proposed via #2079 currently depends on some upstream images, because those in the openshift org are still in-progress.

This issue is to track moving to the correct OCP builds of these containers, and also similar to #2067 we should look at options to move to a staticpod, and perhaps also the MCO instead of the current systemd/script approach.

Ideally I'd like to not block the initial experimental support on this, so I'm raising this as a separate issue to #2060 and we'll address these issues via follow-up PRs

**Comment by stbenjam** [comment]
It looks like all the ironic and image downloading containers are in the releases now:

```
  ironic-image                                  sha256:43dce5dd426ae94ea28b4a074f55336256a05428fbdc631e4391160208b4769e
  ironic-inspector-image                        sha256:c4bd790f1928023425695330eb025098259e7533c78d12b15e992d0c3a9ff59e
  ironic-ipa-downloader                         sha256:29b7428d4a886e4ee7c41128a19d7a0112cd846994ac8c33f0e1c264b41441fa
  ironic-rhcos-downloader                       sha256:281af58fc909a6a45c0669264e6ab95e80365f7df2fc148c086ba83722ab96a7
```

How would we move it to the static pods? I think we want to ensure an ordering - right? Downloading of images needs to happen before Ironic starts, can that be controlled in MCO? Is this something we need to deal with later?

**Comment by dhellmann** [comment]
Those image names are going to be changing shortly. See https://github.com/openshift/release/pull/4623 and https://github.com/openshift/machine-api-operator/pull/370

**Comment by hardys** [comment]
@stbenjam I raised https://github.com/openshift/installer/issues/2251 to investigate how we migrate to staticpods, I'm unclear if MCO Is an option since we only want this to run on the boostrap VM?

I know @cybertron did the work to move the DNS services (which do have a slightly different configuration on the bootstrap VM) to MCO, perhaps he can provide some pointers before we dig into it.

**Comment by cybertron** [comment]
Yeah, we have separate networking service configs for bootstrap, so it should be possible. You can see the change I made to do that in https://github.com/openshift/machine-config-operator/commit/ec7ef01587296e2ba4ab5f21909d3bc88eddee6c

Note in particular that bootstrap pod definitions go in manifests/. We also have some in templates/, but those are for the masters and workers.

As far as ordering, could you do the image download as an initContainer? We use one to generate our config before we start the service. Seems like that would serve your purposes too.

**Comment by stbenjam** [comment]
#2251 sounds more elegant, but I don't think we can get that in this week, right? My understanding is we need to land #2234 before Friday to make 4.2.
